Issue2

elveshka opened the pull request:
Resolve #4

elveshka committed:
Leave thid code for new branch

elveshka committed:
Merge remote-tracking branch 'origin/justDoIt' into issue2

# Conflicts:
#	src/httpServer/Resources/Session.java

elveshka committed:
Merge remote-tracking branch 'origin/justDoIt' into issue2

# Conflicts:
#	src/httpServer/Resources/Session.java

elveshka committed:
Merge from last actual version

elveshka committed:
Add pom.xml .gitignore

elveshka committed:
Change .xml .iml config files

elveshka committed:
Merge branch 'issue2' of github.com:elveshka/java-run-personal-project into issue2

The upstream has a properly congigured project

elveshka committed:
Update target in .iml config

elveshka committed:
Clean code

elveshka committed:
Add dependencies

elveshka committed:
Add empty test class

elveshka committed:
Transport changes

elveshka committed:
Refactor Server class

Main method now accept port via args[]
Change halls variable implementation to LinkedHashSet for order adding
Its important for response schedule in ascending order

elveshka committed:
Change halls field implementation after prev commit

elveshka committed:
Add 2 tests

elveshka committed:
Some formatting

elveshka committed:
Add try/catch construction

elveshka committed:
Fix response body bug and reformat code

elveshka committed:
Fix some exception handling

stereohorse commented on src/test/java/httpServer/MoviesSearchPageTest.java:
```diff
@@ -0,0 +1,84 @@
+        String expected = response.body().string();
+        response.body().close();
+
+        assertThat(expected).isEqualTo(actual);
```
Должно быть наоборот - `assertThat` принимает `actual`.

stereohorse commented on src/main/java/httpServer/Server/Server.java:
```diff
@@ -8,21 +8,30 @@
+                    new InetSocketAddress(
+                            "localhost", Integer.parseInt(args[0])), 0);
+        } catch (NumberFormatException e) {
+            System.out.printf("Wrong port format %s", args[0]);
```
Приложение продолжит работу после этого блока `catch` и переменная `httpserver` будет иметь значение `null`.

stereohorse commented on src/main/java/httpServer/Server/Server.java:
```diff
@@ -8,21 +8,30 @@
     private static final String moviesSearchPage = "/movies/search";
     private static final String ticketsPage = "/tickets";
     private static final String purchasePage = "/purchase";
+    private static HttpServer httpserver;
```
Стараемся по минимуму использовать `static`.
С такой реализацией у нас не получится поднять несколько экземпляров `Server`.

stereohorse commented on src/test/java/httpServer/MoviesSearchPageTest.java:
```diff
@@ -0,0 +1,84 @@
+
+    @BeforeClass
+    public static void initDataBase() {
+        dayProgramming = new DayProgramming(Server.hardcodeBlock());
```
Сейчас тесты ссылаются напрямую на production-данные.
Несмотря на то, что они захардкожены, эти данные все же используются в приложении после запуска.

Тесты должны быть изолированными и независимыми от окружающей среды, иначе они будут хрупкими.
Каким образом можно отрефакторить код, чтобы в тестах можно задавать состояние хранилища в блоке `given`?

elveshka committed:
Swap actual/expected. Always forget about it :D

elveshka committed:
Refactor Server class, change test

Server can set up on different port multiple times now
Uncompleted MovieSearchPageTest test (transport commit)

elveshka committed:
Refactor Server class

Add setter to todayDayProgramming and startServer method. This feature should be carrying about test.

elveshka committed:
Add methods for more readable and flexibility
